lib/buildingsync/model_articulation/facility.rb

```diff
@@ -0,0 +1,493 @@
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+require_relative 'site'
+require_relative '../helpers/os_lib_model_generation_bricr'
+require_relative '../helpers/os_lib_geometry'
+require_relative '../helpers/Model.hvac'
+
+module BuildingSync
+  class Facility
+    include OsLib_ModelGenerationBRICR
+    include OsLib_Geometry
+
+    # initialize
+    def initialize(facility_xml, ns)
+      # code to initialize
+      # an array that contains all the sites
+      @sites = []
+
+      # reading the xml
+      read_xml(facility_xml, ns)
+    end
+
+    # adding a site to the facility
+    def read_xml(facility_xml, ns)
+      # puts facility_xml.to_a
+      facility_xml.elements.each("#{ns}:Sites/#{ns}:Site") do |site_element|
+        @sites.push(Site.new(site_element, ns))
+      end
+    end
+
+    # generating the OpenStudio model based on the imported BuildingSync Data
+    def generate_baseline_osm
+      if @sites.count == 0
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Facility.generate_baseline_osm', 'There are no sites attached to this facility in your BuildingSync file.')
+        raise 'There are no sites attached to this facility in your BuildingSync file.'
+      else if @sites.count > 1
+             OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Facility.generate_baseline_osm', "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file.")
+             raise "There are more than one (#{@sites.count}) sites attached to this facility in your BuildingSync file."
+           else
+             OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.generate_baseline_osm', "Info: There is/are #{@sites.count} sites in this facility.")
+           end
+      end
+      @sites.each(&:generate_baseline_osm)
+
+      create_building_systems(@sites[0].get_model, @sites[0].get_building_template, @sites[0].get_system_type, "Forced Air")
+    end
+
+    def create_building_systems(model, template, system_type, hvac_delivery_type)
+      add_space_type_loads = true
+      add_constructions = true
+      add_elevators = false
+      add_exterior_lights = false
+      onsite_parking_fraction = 1.0
+      kitchen_makeup = "Adjacent"
+      exterior_lighting_zone = "3 - All Other Areas"
+      add_exhaust = true
+      add_swh = true
+      add_hvac = true
+      htg_src = "NaturalGas"
+      clg_src = "Electricity"
+      remove_objects = false
+      add_thermostat = true
+
+      initial_objects = model.getModelObjects.size
+
+      OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.create_building_system', "The building started with #{initial_objects} objects.")
+
+      # Make the standard applier
+      standard = Standard.build(template)
```

Comment:
What is Standard?  Where is it defined?  Is this an OpenStudio Standards Standard?

Reply:
This seems to be duplicating OpenStudio Standards code, if possible it is best to call that code directly to reuse it.

Reply:
This is OpenStudio Standards Standard, so I think we are fine.